prov/verbs/src/verbs_domain.c

```diff
@@ -528,7 +530,7 @@ int fi_ibv_fabric(struct fi_fabric_attr *attr, struct fid_fabric **fabric,
 	(*fabric)->fid.ops = &fi_ibv_fi_ops;
 	(*fabric)->ops = &fi_ibv_ops_fabric;
 
-	fab->all_infos = info;
+	fi_ibv_util_prov.info = info;
```

Comment:
Is there a need to make this global? An earlier commit to verbs provider changed the global variable into a fabric variable.

Also, please add details to the commit message. It would be helpful to have a message that explains why a change is necessary.

Reply:
When it was as the fabric variable and previous version, the following is occurred:
Each time MPI (as an example, but I have a ticket for that problem) invokes fi_getinfo(), fi_getinfo() goes through each provider under the hood, verbs provider fi_getinfo() takes a too much time, because it queries the list of devices and builds the list of `fi_info` entries. When RxM searches the core provider (it produces new invocation of verbs's fi_getinfo() and it queries device list and build fi_info once again). Btw, verbs' fi_fabric does the same, it initializes its own version of fi_info.
The main idea of my changes is reduce the multiple queries of the verbs device list. The same is done for the usnic provider. But my solution is slightly different - I do this when common `fi_ini()` (the beggining of common fi_getinfo()) and I build the list of the `fi_info` to utilize them in verbs' fi_getinfo() and fi_fabric(). In other words, we do this only once.

Reply:
I think most or all of the slowness you describe was fixed recently (the multiple calls to getinfo()). Which fi_ calls specifically are too slow?

Reply:
If the performance of fi_getinfo() is slow: then the caller should only call it once and cache the results, otherwise incorrect behavior can ensue, see commit: 05dd5d4e4c4c4c30131c1de3e

Reply:
Several apps call fi_getinfo() 2-3 times with different hints, in order to separate traffic to different endpoints with different capabilities. This allows them to avoid ordering constraints.  fi_getinfo isn't expected to be called in any fast path operation, but we do need to be concerned with job startup time.

Reply:
@shefty Sure, it is completely the same that I tried to say. MPI calls fi_getinfo() with different hints several times to make choice of provider is more proper. But it takes too much time and isn't acceptable. Just doing this in proposed way in this PR, we avoid the several times of querying of device/port attributes.
Btw, we do the querying of the attributes in fi_fabric. Not sure that we are need it.